### PR TITLE
30 Join Game

### PR DESCRIPTION
**Pull request by mikelax**
Still very much a WIP. 

**NOTE** - the migration in this PR will delete data from all the games, messages, and lounge messages tables to allow for adding the foreign key to game_labels. This was to prevent having to drop this data later on once ready to work on #48.

### CODE REVIEW

File: client/src/components/Games/queries.js [hunk 1]
```diff
@@ -136,6 +148,17 @@ export const onGameMessageUpdated = gql`
   }
 `;
 
+// game players
+
+export const gamePlayersQuery = gql`
+  query gamePlayers($gameId: ID!) {
```
Comment (mikelax) [inline client/src/components/Games/queries.js]: @nazar  I tried to add in a parameter for status, but I couldn't get the syntax to work, was getting an HTTP 400 error.

File: client/src/components/Games/queries.js [hunk 2]
```diff
@@ -42,6 +42,18 @@ export const gameLoungeMessageFields = `
   ${userMetadataFields}
 `;
 
+// game polayers
+
+export const gamePlayerFields = `
+  fragment gamePlayerFields on GamePlayer {
+    id
+    user {
```
Comment (mikelax) [inline client/src/components/Games/queries.js:41]: @nazar  these can be updated once we merge in the ticket you are working on

File: api/schemas/gamePlayer.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+              .tap((gamePlayerMessage) => {
+                pubsub.publish(GAME_PLAYER_UPDATED, {
+                  gamePlayerUpdated: gamePlayerMessage
+                });
+              });
+          });
+      })
+  },
+  Subscription: {
+    gamePlayerAdded: {
```
Comment (mikelax) [inline api/schemas/gamePlayer.js:89]: @nazar  I'll need some help with hooking up the subscription to the players component.

File: client/src/components/Games/View/components/GamePlayers.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+          ))}
+        </div>
+      </React.Fragment>
+    );
+  }
+}
+
+export default compose(
+  graphql(gamePlayersQuery, {
+    options: ({ gameId }) => ({ variables: { gameId, status: ['accepted'] } })
```
Comment (mikelax) [inline client/src/components/Games/View/components/GamePlayers.js]: @nazar  once we get the other syntax error working in the query, then the status here should be updated to also include `'pending'`

File: client/src/components/Games/queries.js (hunk 1 above)
Comment (nazar) [reply client/src/components/Games/queries.js]: @mikelax - did you add the `status` parameter to the [query definition](https://github.com/mikelax/forgingadventures/blob/30-joingame/api/schemas/queries.js#L21) in the API schema? 

I'm thinking maybe

```
gamePlayers(gameId: ID!, status: [String]): [GamePlayer!]
```

I'll have a play with the query

Comment (nazar) [reply client/src/components/Games/queries.js]: yep - that worked

Comment (mikelax) [reply client/src/components/Games/queries.js]: thanks @nazar  I pulled and I see the query now returns, no 400 error. 
One issue though is that it does not appear to respect the values I submit here. 

ie. I think the destructured "default" values I am using [here](https://github.com/mikelax/forgingadventures/pull/49/files#diff-837acce59d7ca64e0718c0c0f8229800R39) are always being used for the query, even if a value is passed in to the `status` parameter.

Comment (nazar) [reply client/src/components/Games/queries.js]: @mikelax the `status` parameter will also have to be defined in the [client query](https://github.com/mikelax/forgingadventures/blob/30-joingame/client/src/components/Games/queries.js#L154-L155).

I tried a manual query from graphiql and it did accept the status when I set it in the query,

Comment (mikelax) [reply client/src/components/Games/queries.js]: @nazar  that worked. thanks!  🚀

Comment (nazar) [reply client/src/components/Games/queries.js]: 🚀  🎆  🚒  👨‍🚒

**Review by mikelax — COMMENTED** [review]
does not include joining game, but displays game players.

**Comment by mikelax** [comment]
Addresses #30

**Pull request merged**